src/library/scala/collection/Iterable.scala

```diff
@@ -85,12 +100,12 @@ trait IterableOps[+A, +CC[_], +C] extends Any with IterableOnce[A] with Iterable
     * the elements of the resulting collections). In other words, this methods defines
     * the evaluation model of the collection.
     */
-  protected[this] def fromSpecificIterable(coll: Iterable[A @uncheckedVariance]): C
+  protected def fromSpecificIterable(coll: Iterable[A @uncheckedVariance]): C
```

Comment:
Maybe you can also add a comment here, and to `newSpecificBuilder` (restrictions on how this method can be used)